fed/mosaic-application/src/main/java/org/eclipse/mosaic/fed/application/ambassador/simulation/electric/providers/ChargingStationIndex.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2025 Fraunhofer FOKUS and others. All rights reserved.
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contact: mosaic@fokus.fraunhofer.de
+ */
+
+package org.eclipse.mosaic.fed.application.ambassador.simulation.electric.providers;
+
+import org.eclipse.mosaic.fed.application.ambassador.simulation.electric.objects.ChargingStationObject;
+import org.eclipse.mosaic.fed.application.ambassador.simulation.perception.index.objects.SpatialObjectAdapter;
+import org.eclipse.mosaic.lib.geo.CartesianPoint;
+import org.eclipse.mosaic.lib.geo.GeoCircle;
+import org.eclipse.mosaic.lib.geo.GeoPoint;
+import org.eclipse.mosaic.lib.objects.electricity.ChargingStationData;
+import org.eclipse.mosaic.lib.spatial.KdTree;
+import org.eclipse.mosaic.lib.spatial.SpatialTreeTraverser;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A {@link ChargingStationIndex} holds Charging Stations in a tree structure, sorted by their position.
+ * The tree is initialized and gets updated lazily when a search is performed.
+ */
+public class ChargingStationIndex {
+    private final int bucketSize;
+
+    /**
+     * Stores {@link ChargingStationObject}s for fast removal and position update.
+     */
+    final Map<String, ChargingStationObject> indexedChargingStations = new HashMap<>();
+
+    private KdTree<ChargingStationObject> chargingStationTree;
+
+    private SpatialTreeTraverser.InRadius<ChargingStationObject> treeTraverser;
+
+    private boolean needsTreeUpdate = false;
+
+    public ChargingStationIndex(int bucketSize) {
+        this.bucketSize = bucketSize;
+    }
+
+    /**
+     * Inits a {@link ChargingStationIndex} with default bucket size of 20.
+     * Bucket size describes the item capacity of one tree node.
+     */
+    public ChargingStationIndex() {
+        this.bucketSize = 20;
+    }
+
+    /**
+     * Adds a Charging Station to the tree.
+     * Be sure to add {@link ChargingStationData} using updateChargingStation(ChargingStationData chargingStationData).
```

Comment:
please also reference to methods using the {@link} syntax. You can use '#' to reference methods